pkgs/applications/virtualization/colima/default.nix

```
@@ -1,54 +1,66 @@
 { lib
-, buildGo118Module
+, buildGoModule
 , fetchFromGitHub
 , installShellFiles
 , lima
 , makeWrapper
+, qemu
+, testers
+, colima
+, runCommand
 }:
 
-buildGo118Module rec {
+buildGoModule rec {
   pname = "colima";
-  version = "0.4.2";
+  version = "0.4.4";
 
   src = fetchFromGitHub {
     owner = "abiosoft";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-66nKH5jxTzLB9bg2lH1E8Cc0GZ6C/N/+yPYhCVEKOBY=";
-
+    sha256 = "bSBaSS+rVkFqTSdyegdE/F0X5u7yvF/nHslAO3xgD6I=";
     # We need the git revision
     leaveDotGit = true;
     postFetch = ''
-      git -C $out rev-parse HEAD > $out/.git-revision
+      git -C $out rev-parse --short HEAD > $out/.git-revision
       rm -rf $out/.git
     '';
   };
 
   nativeBuildInputs = [ installShellFiles makeWrapper ];
 
-  vendorSha256 = "sha256-91Ex3RPWxOHyZcR3Bo+bRdDAFw2mEGiC/uNKjdX2kuw=";
+  vendorSha256 = "jDzDwK7qA9lKP8CfkKzfooPDrHuHI4OpiLXmX9vOpOg=";
 
-  doCheck = false;
+  CGO_ENABLED = 1;
 
   preConfigure = ''
-    ldflags="-X github.com/abiosoft/colima/config.appVersion=${version}
-              -X github.com/abiosoft/colima/config.revision=$(cat .git-revision)"
+    ldflags="-s -w -X github.com/abiosoft/colima/config.appVersion=${version} \
+    -X github.com/abiosoft/colima/config.revision=$(cat .git-revision)"
   '';
 
+  subPackages = [ "cmd/colima" ];
+
   postInstall = ''
     wrapProgram $out/bin/colima \
-      --prefix PATH : ${lib.makeBinPath [ lima ]}
+      --prefix PATH : ${lib.makeBinPath [ lima qemu ]}
 
     installShellCompletion --cmd colima \
       --bash <($out/bin/colima completion bash) \
       --fish <($out/bin/colima completion fish) \
       --zsh <($out/bin/colima completion zsh)
   '';
 
+  passthru.tests.version = testers.testVersion {
+    package = colima;
+    command = "HOME=$(mktemp -d) colima version";
+  };
+
   meta = with lib; {
-    description = "Container runtimes on MacOS with minimal setup";
+    description = "Container runtimes on MacOS (and Linux) with minimal setup";
     homepage = "https://github.com/abiosoft/colima";
     license = licenses.mit;
-    maintainers = with maintainers; [ aaschmid ];
+    mainProgram = "colima";
+    platforms = [ "aarch64-linux" "aarch64-darwin" "x86_64-linux" "x86_64-darwin" ];
```

```suggestion
    platforms = with platforms; linux ++ darwin;
```

or something like that

---

@Atemu 
Yes, I usually did that but from other reviews I got the feedback that I should use the above because
`platforms.linux ++ platforms.darwin = [ "aarch64-linux" "armv5tel-linux" "armv6l-linux" "armv7a-linux" "armv7l-linux" "i686-linux" "m68k-linux" "mipsel-linux" "mips64el-linux" "powerpc64-linux" "powerpc64le-linux" "riscv32-linux" "riscv64-linux" "s390-linux" "s390x-linux" "x86_64-linux" "x86_64-darwin" "i686-darwin" "aarch64-darwin" "armv7a-darwin" ]` results in a long list and I don't know if every platform is supported.

Guess it makes more sense to add supported platforms in the future than removing unsupported platforms due to bugs. What do you think? Is there a general consensus?

---

and linux + darwin is the default so we can drop it.

> but from other reviews I got the feedback that I should use the above because

thats not really how we do things.

> results in a long list and I don't know if every platform is supported.

I can guarantee you that it isn't but thats okay. We don't want to touch the meta sections of thousands of packages if a new architecture gets great support.

Generally this is mostly about if a package is supported on linux or darwin and x86_64 or aarch64. All the arm variants, m68k, mipsel or s390 don't really matter especially that we don't have hydra support for them.

---

@SuperSandro2000 Ok, I removed the meta.platforms attribute again:+1:.

---

@tricktron FYI, you didn't.

---

@Atemu Did you see my latest commit 10 hours ago https://github.com/NixOS/nixpkgs/commit/62b5ad1363dc5a70edf1a630ca0939ab2b0c6a2f?